smolyay/samples.py

```diff
@@ -508,3 +511,580 @@ def _create(self):
+
+    @MultidimensionalPointSet.domain.setter
+    def domain(self, value):
+        domain = numpy.sort(numpy.array(value, ndmin=2), axis=1)
+        if domain.ndim != 2 or domain.shape[1] != 2:
+            raise TypeError("Domain must have size (num_dimensions, 2)")
+        if numpy.any(domain[:, 0] >= domain[:, 1]):
+            raise ValueError("Lower bound must be less than upper bound")
+        if not numpy.array_equal(self._domain, domain):
+            self._domain = domain
+            self._valid_cache = False
+
+    @property
+    def num_points(self):
+        """int: Number of points"""
+        return self._num_points
+
+    @num_points.setter
+    def num_points(self, value):
+        num_points = int(value)
+        if self._num_points != num_points:
+            self._num_points = num_points
+            self._valid_cache = False
+
+    @property
+    def seed(self):
+        """int: random seed for generating points"""
+        return self._seed
+
+    @seed.setter
+    def seed(self, value):
+        value = int(value)
+        if self._seed != value:
+            self._seed = value
+            self._valid_cache = False
+
+
+class UniformRandomPointSet(RandomPointSet):
+    """Points drawn from a uniform random distribution."""
+
+    def _create(self):
+        """Generate a set of random points
+
+        Generates uniformly distributed random points
+        """
+        lower_bounds = self.domain[:, 0]
+        upper_bounds = self.domain[:, 1]
+        num_dimensions = len(lower_bounds)
+        p_gen = numpy.random.default_rng(seed=self.seed).uniform(
+            size=(self.num_points, num_dimensions)
+        )
+        self._points = scipy.stats.qmc.scale(p_gen, lower_bounds, upper_bounds)
+
+
+class _QMCRandomPointSet(RandomPointSet):
+    """Points drawn randomly using a QMCEngine
+
+    This random point set relies on the QMCEngine objects in
+    the scipy.stats.qmc module to generate points.
+
+    In addition to the domain, number of points, and the seed
+    to generate the points, all QMCEngines have 2 additional
+    optional parameters, ``scramble`` and ``optimization``.
+
+    Parameters
+    ----------
+    domain: list
+        domain of the random points
+
+    num_points: int
+        number of points to generate
+
+    seed: int
+        seed for generating the random points
+
+    scramble : bool
+        Default True. Applies centering to Latin Hypercube points, Owen
+        scrambling to Halton points, and LMS+shift scrambling to Sobol points.
+
+    optimization : {None, "random-cd", "lloyd"}
+        Default None. If "random-cd" the coordinates of points are adjusted to
+        lower the centered discrepancy. If "lloyd", adjust points using a
+        Lloyd-Max algorithm to encourage even spacing.
+    """
+
+    def __init__(self, domain, num_points, seed, scramble=True, optimization=None):
+        super().__init__(domain, num_points, seed)
+        self._scramble = True
```

Comment:
Default to `None` to force the setter to run?
```suggestion
        self._scramble = None
```